rest/route/host_agent.go

```diff
@@ -418,7 +418,9 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 			"task_version":       nextTask.Version,
 		}
 		if err != nil {
-			grip.Alert(message.WrapError(err, errMsg))
+			if !errors.Is(err, context.Canceled) && ctx.Err() == nil {
```

Comment:
Is this really an alert condition at all? I feel like the alert cases in this function really aren't worth alerting on each one. Lowering them to error seems more appropriate and they would still appear in our error rate.